integration_tests/src/lib.rs

```diff
@@ -24,6 +25,305 @@ mod test {
     use tokio::sync::mpsc;
     use uuid::*;
 
+    trait TestDownstairsDataset: Sized {
+        /// Create a new dataset for a region of [`size`] bytes
+        fn new(size: u64) -> Result<Self>;
+
+        /// The path where the Downstairs should run
+        fn path_buf(&self) -> Result<PathBuf>;
+
+        /// Returns true if a snapshot exists
+        fn snapshot_exists(&self, snapshot_name: &str) -> Result<bool>;
+
+        /// Should the Downstairs be stopped during the drop of this object?
+        ///
+        /// Note: if this returns true, a multi-threaded tokio runtime is
+        /// required!
+        fn stop_downstairs_during_drop(&self) -> bool;
+    }
+
+    #[cfg(target_os = "illumos")]
+    mod zfs_storage {
+        use super::*;
+        use std::fs::OpenOptions;
+        use std::process::Command;
+
+        #[derive(Debug)]
+        pub struct Zpool {
+            name: String,
+        }
+
+        impl Zpool {
+            pub fn new(name: String, path: String) -> Result<Self> {
+                let output = Command::new("pfexec")
+                    .args(["zpool", "create", &name, &path])
+                    .output()?;
+
+                if !output.status.success() {
+                    bail!(
+                        "zpool create failed {} {}",
+                        String::from_utf8_lossy(&output.stdout),
+                        String::from_utf8_lossy(&output.stderr),
+                    );
+                }
+
+                Ok(Zpool { name })
+            }
+        }
+
+        impl Drop for Zpool {
+            fn drop(&mut self) {
+                let output = Command::new("pfexec")
+                    .args(["zpool", "list", &self.name])
+                    .output()
+                    .unwrap();
+
+                if output.status.success() {
```

Comment:
Nit: should we panic here as well, or are there cases where pools will be destroyed without our knowledge?